cypress/e2e/CMS/$Article/index.js

```diff
@@ -0,0 +1,152 @@
+    cy.get(`h2[id='article-heading1'] p:contains('${title}')`).should('be.visible')
+})
+
+And('{int} description reads {string}', (contentText) => {
+    cy.get(`div[class='medium-9 columns contentzone has-section-nav'] p:contains('${contentText}')`).should('be.visible')
+})
+
+And('public use text has a link {string} with href {string}', (linkText, linkUrl) => {
+    cy.get(`div[class='public-use'] a:contains('${linkText}')`).should('be.visible').and('have.attr', 'href', linkUrl)
+})
+
+And('{string} section is not displayed', (section) => {
+    cy.get(`a:contains("${section}")`).should('not.exist')
+})
+
+let imageSrc;
+And('user selects {int} Lead Image from the list of images', (num) => {
+    cy.get('span:contains("Lead Image")').parent().as('imageUpload').click()
+    cy.get('input[name="field_image_article_entity_browser_entity_browser"]').click({ force: true })
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find("input[id^='edit-entity-browser-select-media']").eq(num - 1).check()
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find("input[id='edit-submit'][value='Select image']").click({ force: true })
+}); 
+And('user remembers the source of selected lead image for further verification', () => {
+    cy.get('details img').then($el => {
+        imageSrc = $el[0].getAttribute('src')
+    })
+})
+
+let imageSrc1;
+And('user selects {int} Promotional Image from the list of images', (num) => {
+    cy.get('span:contains("Promotional Image")').parent().as('imageUpload').click()
+    cy.get('input[name="field_image_promotional_entity_browser_entity_browser"]').click({ force: true })
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find("input[id^='edit-entity-browser-select-media']").eq(num - 1).check()
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find("input[id='edit-submit'][value='Select image']").click({ force: true })
+})
+
+And('user remembers the source of selected promo image for further verification', () => {
+    cy.get('details img').then($el => {
+        imageSrc1 = $el[0].getAttribute('src')
+    })
+})
+
+And('user selects {string} checkbox', (dateDisplay) => {
+    cy.get(`[class='fieldset-wrapper'] label:contains("${dateDisplay}")`).parent().find('input.form-checkbox').check({ force: true })
+})
+
+And("{string} date is displaying today's date", (date) => {
+    cy.get('ul.clearfix li strong').should('be.visible').and('include.text', date)
+})
+
+And('the lead image is matching the earlier selected image', () => {
+    cy.get('.centered-element > img').then($el=>{
+    const source = $el[0].getAttribute('src');
+    const actSrc = source.replace(/\?itok=[\S]+/,'').replace(/^(.*?)\/public/,'')
+    const expectedSrc = (imageSrc.replace(/\?itok=[\S]+/,'')).replace(/^(.*?)\/public/,'')
+    expect(actSrc).to.include(expectedSrc);
+    })
+    });
+    
+
+And('user clicks on dropdown button toggle to view all Related Resources types', () => {
```

Comment:
all of the steps related to adding a related resource are copy pasted from common - these are all duplicates, remove them from here.

The reason why we added them in common folder is that we do not duplicate code over and over